docs/how-tos/hello-world-module.md

```diff
@@ -0,0 +1,304 @@
+   - Enable cloud build: `No`
+   - Register module: `No`
+
+1. Hit your Enter key and the generator will generate a folder called <file>hello-world</file> containing stub files for your modular camera.
+
+## Implement the API
+
+Edit the stub files to implement your test script in a way that works with the camera API:
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+1. Open the <file>hello-world/src/main.py</file> file in your code editing program.
+
+1. Add the following to the list of imports at the top of <file>main.py</file>:
+
+    ```python {class="line-numbers linkable-line-numbers"}
+    from viam.media.utils.pil import pil_to_viam_image
+    from viam.media.video import CameraMimeType
+    from viam.utils import struct_to_dict
+    from PIL import Image
+    ```
+
+1. Instead of hard-coding the path to the image into the module itself, make the path a configurable attribute so you or other users of the module can easily set a different path.
+   Add the following lines to the `reconfigure()` function definition:
+
+    ```python {class="line-numbers"}
+    attrs = struct_to_dict(config.attributes)
+    self.image_path = str(attrs.get("image_path"))
+    ```
+
+1. Since the camera won't work unless the user configures an `image_path` attribute, add the following code to the `validate()` function to throw an error if `image_path` isn't configured:
+
+    ```python {class="line-numbers linkable-line-numbers"}
+    # Check that a path to get an image was configured
+    fields = config.attributes.fields
+    if not "image_path" in fields:
+        raise Exception("Missing image_path attribute.")
+    elif not fields["image_path"].HasField("string_value"):
+        raise Exception("image_path must be a string.")
+    ```
+
+1. The module generator created a stub for the `get_image()` function we want to implement:
+
+   ```python {class="line-numbers linkable-line-numbers" data-start="79" }
+    async def get_image(
+        self,
+        mime_type: str = "",
+        *,
+        extra: Optional[Dict[str, Any]] = None,
+        timeout: Optional[float] = None,
+        **kwargs
+    ) -> ViamImage:
+        raise NotImplementedError()
+   ```
+
+   You need to replace `raise NotImplementedError()` with code to actually implement the method:
+
+   ```python {class="line-numbers linkable-line-numbers" data-start="86" }
+   ) -> ViamImage:
+       img = Image.open(self.image_path)
+       return pil_to_viam_image(img, CameraMimeType.JPEG)
+   ```
+
+    You can leave the rest of the functions not implemented, because this module is not meant to return a point cloud (`get_point_cloud()`), and does not need to return multiple images simultaneously (`get_images()`).
+
+1. Open <file>requirements.txt</file>.
+   Add the following line:
+
+   ```text
+   Pillow
+   ```
+
+1. Create a virtual Python environment with necessary packages by running the setup file:
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+sh setup.sh
+```
```

Comment:
This looks like it worked for the other oens
```suggestion
   ```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
   sh setup.sh
   ```
```

Reply:
Oop yep